### ISSUE TITLE
Secure the build requests

### ISSUE THREAD

**Issue by zepumph** [phetsims/perennial#39]
Currently the URL outputted from grunt deploy-production/rc is all you need to start the build. This is because it is a GET request, and the authentication is housed in the URL itself. This is quite unsafe. We realized this when a url was pasted into the skype chat, and 12 hours later a Skype bot had started a build. 

Two ways to secure the build requests:
- [ ] Make the request a POST.
- [ ] Put authentication into the headers, basic authentication is probably all we need.

**Comment by jonathanolson** [comment]
Both sound good to me.

**Comment by jonathanolson** [comment]
POST-data seems sufficient. We'll also want to potentially send build-request versions, so we can differentiate requests in the future if needed.

**Comment by mattpen** [comment]
As a reminder to myself, this required moving the varnish routing for build-server before the rules that ignore caching of non-GET/HEAD requests.

**Comment by mattpen** [comment]
I have implemented a POST API in the build-server and changed chipper such that deploy-rc and deploy-production now send POST requests instead of query parameters.  I have tested both rc and prod deploys with chains on ox-dev.

We will also need to update rosetta's call to the build-server to use post requests instead of query params.

I have left the query parameter API in place to support old versions of chipper.  Once https://github.com/phetsims/chipper/commit/504a620f9568205cf0b014600d99ce830e98ec78 has been pulled into all sims we can remove it.
